targets/targets.json

```diff
@@ -3613,7 +3613,7 @@
         "inherits": ["Target"],
         "detect_code": ["4600"],
         "extra_labels": ["Realtek", "AMEBA", "RTL8195A"],
-        "macros": ["__RTL8195A__","CONFIG_PLATFORM_8195A","CONFIG_MBED_ENABLED","PLATFORM_CMSIS_RTOS"],
```

Comment:
REALTEK_AMEBA provides their own implementation of HardFault handler as binary. So, disabling Mbed-OS fault handling.

Reply:
Thanks for the details, they should be part of the commit message to have them available not only on github

`MBED_FAULT_HANDLER_DISABLED` vs `CONFIG_MBED_ENABLED` (enabled or disabled might be confusing, so one of these should be fixed?). Can we have this fault handler enabled by default, and only removed if not used (first comes to my mind using Target from targets to define it , or another way to do this) ?

Reply:
@0xc0170 - Yes, Currently this is enabled by default and is disabled only if you specify MBED_FAULT_HANDLER_DISABLED flag. targets.json can be used to disable this on specific targets if not required. I think that matches your expectations?

Reply:
@SenRamakri Could you make this configurable through the configuration system? Currently it's a macro, and we would like to eliminate macros going forward.

Reply:
```json
{                                                                                                        
    "name": "rtos",                                                                                      
    "config": {                                                                                          
        "present": 1,                                                                                    
        "detailed-fault-handler": {                                                                      
            "help": "Configures the detail of the fault handler",                                        
            "value": true                                                                                
        }                                                                                                
    },                                                                                                   
    "target_overrides" : {                                                                               
        "K64F": {                                                                                        
            "detailed-fault-handler": false                                                              
        }                                                                                                
    }                                                                                                    
}
 ```

Reply:
@0xc0170 - Looks like we cannot use Json based config system for this as the Config defines are not visible in .S files for IAR toolchain. @theotherjimmy and I tried to work through this but couldn't really make it work for IAR due to some differences in how IAR preprocessing works. So, reverting back to old mechanism, unfortunately.